### PR TITLE
SAK-29457 Portal: Add warning about cookie use

### PR DESCRIPTION
**Pull request by mcarropellicer**
SAK-29457 Portal: Add warning about cookie use

### CODE REVIEW

**Comment by buckett** [comment]
Is there any configuration to turn this on/off?

**Comment by mcarropellicer** [comment]
Not yet but it's pretty easy to include :dancers:, i'll do it

**Comment by mcarropellicer** [comment]
Now it's configurable by property, TRUE by default.

**Comment by clhedrick** [comment]
The text is pretty uninformative. We use cookies to maintain state. Without them as far as I know Sakai doesn't work. "to get the best experience" is typical commercial language designed not to say anything.

**Comment by mcarropellicer** [comment]
Feel free to discuss with the authors and the community to find a better text, thanks for your suggestion.

**Comment by ottenhoff** [comment]
Can we set the default to FALSE as there are many more institutions in regions that do not require a cookie warning?

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 1]
```diff
@@ -57,6 +57,14 @@
 # DEFAULT: true
 # portal.use.tutorial=false
 
+# SAK-29457
+# Enable/disable the cookie policy warning
+# DEFAULT: true
```
Comment (steveswinsburg) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties]: Is default true or false?

**Comment by steveswinsburg** [comment]
I'm not convinced this is required at all, given the three cookies we create.

From: http://ec.europa.eu/ipg/basics/legal/cookies/index_en.htm

Cookies clearly exempt from consent according to the EU advisory body on data protection include:

* user‑input cookies (session-id) such as first‑party cookies to keep track of the user's input when filling online forms, shopping carts, etc., for the duration of a session or persistent cookies limited to a few hours in some cases
* authentication cookies, to identify the user once he has logged in, for the duration of a session
user‑centric security cookies, used to detect authentication abuses, for a limited persistent duration
* multimedia content player cookies, used to store technical data to play back video or audio content, for the duration of a session
* load‑balancing cookies, for the duration of session
* user‑interface customisation cookies such as language or font preferences, for the duration of a session (or slightly longer)
* third‑party social plug‑in content‑sharing cookies, for logged‑in members of a social network.

See also the _Use on EUROPA_ section.

All of our cookies are first party session cookies and DO NOT require informed consent.

So I am -1 on this at the moment. At the very least, this must default to false as it is only for EU jurisdictions.

**Comment by clhedrick** [comment]
This sounds right to me

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 2]
```diff
@@ -57,6 +57,14 @@
...
+# portal.cookie.policy.warning.enabled = false
+#
```
Comment (steveswinsburg) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties]: The doc says default true but the property is marked as default false.

**Comment by mcarropellicer** [comment]
Changes done, now the default value is false.

**Comment by steveswinsburg** [comment]
Thanks. Though as above, is this even required? The cookies we use make us exempt.

**Comment by buckett** [comment]
I think it's a little grey, we set a cookie when someone first visits the service even if they are using functionality that doesn't require a session. For example if you just want to view the announcements on the !gateway site the functionality doesn't require a cookie, but we still end up setting one because of the way we use sessions.

https://ico.org.uk/for-organisations/guide-to-pecr/cookies-and-similar-technologies/

**Comment by ottenhoff** [comment]
I'm sure there are lawyers at European institutions that have decided this warning is necessary.... those institutions will need to implement this warning. I agree it should be disabled by default.

**Comment by mcarropellicer** [comment]
We're exempt for this type of cookies in Spain, but this is good to have in Sakai because exists a Google Analytics integration.

Exempt:
Technical, Configuration, Security

Mandatory warning:
Analytics, Tracking, Advertising

Miguel

**Comment by mcarropellicer** [comment]
Sorry for bothering, can this be merged into Trunk? It's disabled by default.

Thank you, Miguel

**Comment by sinmsinm** [comment]
LGTM :+1:

**Pull request merged**